catalog/dags/data_refresh/promote_table.py

```diff
@@ -0,0 +1,99 @@
+"""
+# Promote Table
+
+This file contains TaskGroups related to promoting the API DB tables as part of the data refresh.
+
+In this TaskGroup, indices and constraints from the original table are applied to the temp table before dropping the original table and "promoting" the temp table by renaming it.
+"""
+
+import logging
+from dataclasses import asdict
+
+from airflow.decorators import task, task_group
+
+from common.constants import POSTGRES_API_CONN_IDS, Environment
+from common.sql import run_sql
+from data_refresh import queries
+from data_refresh.data_refresh_types import DataRefreshConfig
+from data_refresh.remap_table_constraints import remap_table_constraints_to_table
+from data_refresh.remap_table_indices import TableIndex, remap_table_indices_to_table
+
+
+logger = logging.getLogger(__name__)
+
+
+@task
+def get_go_live_query(
+    table_name: str, temp_table_name: str, index_mapping: list[TableIndex]
+):
+    """Get the query for replacing the original media table with the new temp table."""
+
+    # The remapped indices on the temp table all have temporary names to avoid
+    # naming collisions with the indices on the original table. Prepare the ALTER
+    # TABLE statements for restoring them to their original names once the original
+    # table is dropped.
+    restore_index_names = ("\n        ").join(
+        [
+            queries.RENAME_INDEX_QUERY.format(
+                old_name=index.temp_index_name, new_name=index.index_name
+            )
+            for index in index_mapping
+        ]
+    )
+
+    return queries.GO_LIVE_QUERY.format(
+        table_name=table_name,
+        temp_table_name=temp_table_name,
+        restore_index_names=restore_index_names,
+    )
+
+
+@task_group(group_id="promote_table")
```

Comment:
It looks like [`group_id` defaults to the function name](https://github.com/apache/airflow/blob/278dd31228f73f926eff68124374d0db1f4d70b8/airflow/decorators/task_group.py#L75), so we don't need to do anything more than `@task_group` both here and below!